### PR TITLE
New DeathScreen

### PR DESCRIPTION
**Pull request by devildrake**
Established new DeathScreen assets
The new hierarchy changes for DeathScreen to be copied on all scenes can be found in DeathScreen.prefab
Modified code, now transitions will use CanvasFader if referenced, CanvasFader is useful to make transitions work with centralized code
Established Level1_NewDeathScreen, when all approves are issued this scene is to be deleted.
If this PR is approved, the UI changes will become necesary on all scenes (A list of steps could be a good idea)

Also added a few adjustments (Code only to the UI, now when switching scale only affects X, all abilities now wave between high and low alpha like the switch ability always did)

### CODE REVIEW

File: Penteract/Assets/Scripts/GameOverUIController.cpp [hunk 1]
```diff
@@ -31,23 +38,40 @@ void GameOverUIController::Start() {
...
+			canvasFader = GET_SCRIPT(canvasFaderObj, CanvasFader);
+		}
+	}
+
+
 	if (playAgainGO) {
 		playAgainButtonImage = playAgainGO->GetComponent<ComponentImage>();
-		playAgainButtonText = playAgainGO->GetChildren()[0]->GetComponent < ComponentText>();
+		playAgainButtonText = playAgainGO->GetChildren()[HIERARCHY_INDEX_PLAY_AGAIN_TEXT_]->GetComponent < ComponentText>();
 	}
```
Comment (FrancPS) [inline Penteract/Assets/Scripts/GameOverUIController.cpp:67]: Crashed here when pressing Play Again
Sin título

**Comment by devildrake** [comment]
> Check again the buttons, color and sizes. The text should be duplicated to add a shadow efect. Check the ExitGameButton image "color" in the inspector, it should be white.
> 
> I can help you with the details of this, as I've been working with Jaime in the Start scene about these.

Duplicated text for shadowin WILL be and issue, I suggest leaving that for another PR.
As for buttons sizes, I used the layout as a reference, feel free to tweak as you see fit, I won't be really available this weekend

**Review by GBMiro — COMMENTED** [review]
Looks good. I'll test it this afternoon and give the approval

**Comment by FrancPS** [comment]
Fixed this:
Sin título


Another iteration of the menus will be needed later to include the correct audio of the buttons.
This iteration will need some changes included in StartScene.cpp, so this PR can be accepted for now.

File: Penteract/Assets/Scripts/GameOverUIController.cpp [hunk 2]
```diff
@@ -68,7 +104,22 @@ void GameOverUIController::Start() {
 }
 
 void GameOverUIController::Update() {
-	if (!inOutPlayer || !loopPlayer || !vibrationPlayer) return;
+	if (!inOutPlayer || !loopPlayer || !vibrationPlayer || !outInPlayer) return;
+
+	if (state != GameOverState::FADE_OUT && state != GameOverState::OFFLINE) {
+		if (canvasFader) {
+			if (canvasFader->IsPlaying()) {
+				Debug::Log("CanvasFaderPlaying");
```
[the remark this answers is not in the document]
Comment (FrancPS) [reply Penteract/Assets/Scripts/GameOverUIController.cpp:112]: XD

**Comment by FrancPS** [comment]
> 
> 
> I have the feeling that buttons are oversized. When buttons are hovered, the shadow disappears, is that intended?
> 
> I don't have the Drive document reference, but I'm assuming everything is ok, so, approved.

Yep, nope, you are right.
All of this will be reedited with the audio fixes, as new prefabs will be available.

**Review by magalenyo — APPROVED** [review]

**Review by GBMiro — APPROVED** [review]

**Pull request merged**